src/pages/components/Overlay.qml

```diff
@@ -415,15 +419,23 @@ PanelBackground {
 
             onStatusChanged: {
                 if (activeWebPage && status == PageStatus.Active) {
-                    activeWebPage.grabToFile()
+                    webView.privateMode ? activeWebPage.grabThumbnail() : activeWebPage.grabToFile()
                 }
             }
 
             Browser.TabView {
                 model: webView.tabModel
                 portrait: tabPage.isPortrait
+                privateMode: webView.privateMode
 
                 onHide: pageStack.pop()
+
+                onPrivateModeChanged: {
+                    webView.privateMode = privateMode
+                    tabPage.activeTabIndex =  webView.tabModel.activeTabIndex
+                    tabPage.activeWebPage = webView.contentItem
```

Comment:
Fixed through 77d1ffb0098f7eb9a7d03514c5b146edc6fb8af0 and I agree that we should stay in  tabview after closing all tabs.